03-May-2021 18:14:51.286 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 18:14:51.286 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 18:14:51.288 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@31c269fd')
03-May-2021 18:15:32.842 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
03-May-2021 18:15:32.842 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
03-May-2021 18:17:05.853 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 18:17:05.853 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 18:17:05.855 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@6e33c391')
03-May-2021 18:17:12.301 SEVERE [http-nio-8080-exec-2] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [An exception occurred processing [group11.jsp] at line [74]

71:                 scores.add(String.format("%.2f", Double.parseDouble(record.get(0))));
72:                 temp.add(record.get(1));
73:                 title.add(record.get(2));
74:                 url.add(record.get(3));
75:                 last_modification_date.add(record.get(4));
76:                 size.add(record.get(5));
77: 


Stacktrace:] with root cause
	java.lang.ArrayIndexOutOfBoundsException: Index 3 out of bounds for length 3
		at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4164)
		at org.apache.jsp.group11_jsp._jspService(group11_jsp.java:203)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:71)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:467)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:353)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:870)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1696)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:832)
03-May-2021 18:17:23.369 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [An exception occurred processing [group11.jsp] at line [214]

211:                         List<String> search_date = new ArrayList<>();
212:                         if (sessionExist && s != null && !s.equals("") && session.getAttribute("name") != null && session.getAttribute("search_date") != null)
213:                             search_history = (ArrayList<String>) session.getAttribute("name");
214:                         search_date = (ArrayList<String>) session.getAttribute("search_date");
215:                         for (int i = search_history.size() - 1; i >= 0; i--) {
216:                     %>
217:                     <div class="row">


Stacktrace:] with root cause
	java.lang.IllegalStateException: getAttribute: Session already invalidated
		at org.apache.catalina.session.StandardSession.getAttribute(StandardSession.java:1128)
		at org.apache.catalina.session.StandardSessionFacade.getAttribute(StandardSessionFacade.java:102)
		at org.apache.jsp.group11_jsp._jspService(group11_jsp.java:412)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:71)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:467)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:353)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:870)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1696)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:832)
03-May-2021 18:17:31.253 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [java.lang.ArrayIndexOutOfBoundsException] with root cause
	java.lang.ArrayIndexOutOfBoundsException
03-May-2021 18:17:41.878 SEVERE [http-nio-8080-exec-7] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [An exception occurred processing [group11.jsp] at line [214]

211:                         List<String> search_date = new ArrayList<>();
212:                         if (sessionExist && s != null && !s.equals("") && session.getAttribute("name") != null && session.getAttribute("search_date") != null)
213:                             search_history = (ArrayList<String>) session.getAttribute("name");
214:                         search_date = (ArrayList<String>) session.getAttribute("search_date");
215:                         for (int i = search_history.size() - 1; i >= 0; i--) {
216:                     %>
217:                     <div class="row">


Stacktrace:] with root cause
	java.lang.IllegalStateException: getAttribute: Session already invalidated
		at org.apache.catalina.session.StandardSession.getAttribute(StandardSession.java:1128)
		at org.apache.catalina.session.StandardSessionFacade.getAttribute(StandardSessionFacade.java:102)
		at org.apache.jsp.group11_jsp._jspService(group11_jsp.java:412)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:71)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:467)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:353)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:870)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1696)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:832)
03-May-2021 18:17:48.353 SEVERE [http-nio-8080-exec-10] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [java.lang.ArrayIndexOutOfBoundsException] with root cause
	java.lang.ArrayIndexOutOfBoundsException
03-May-2021 18:18:01.736 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
03-May-2021 18:18:01.736 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
03-May-2021 18:18:55.955 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 18:18:55.955 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 18:18:55.957 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@6e33c391')
03-May-2021 18:21:51.549 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
03-May-2021 18:21:51.549 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
03-May-2021 18:22:25.097 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 18:22:25.098 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 18:22:25.100 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@1e886a5b')
03-May-2021 18:22:36.652 SEVERE [http-nio-8080-exec-2] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [An exception occurred processing [group11.jsp] at line [74]

71:                 scores.add(String.format("%.2f", Double.parseDouble(record.get(0))));
72:                 temp.add(record.get(1));
73:                 title.add(record.get(2));
74:                 url.add(record.get(3));
75:                 last_modification_date.add(record.get(4));
76:                 size.add(record.get(5));
77: 


Stacktrace:] with root cause
	java.lang.ArrayIndexOutOfBoundsException: Index 3 out of bounds for length 3
		at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4164)
		at org.apache.jsp.group11_jsp._jspService(group11_jsp.java:203)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:71)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:467)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:770)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:353)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:870)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1696)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:832)
03-May-2021 19:02:43.395 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/example] threw exception [java.lang.ArrayIndexOutOfBoundsException] with root cause
	java.lang.ArrayIndexOutOfBoundsException
03-May-2021 19:03:07.612 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
03-May-2021 19:03:07.612 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
03-May-2021 19:04:01.296 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 19:04:01.297 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 19:04:01.299 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@1e886a5b')
03-May-2021 20:11:58.693 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
03-May-2021 20:11:58.694 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
03-May-2021 20:14:39.000 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
03-May-2021 20:14:39.000 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
03-May-2021 20:14:39.003 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@6e33c391')
